src/methods/join.js

```diff
@@ -445,10 +458,31 @@ export class JoinImplementation {
     let empty_effects = construct_empty_effects(realm);
     let v = realm.intrinsics.empty;
     if (c.consequent instanceof ReturnCompletion) {
-      let pnc = new PossiblyNormalCompletion(v, c.joinCondition, v, empty_effects, c.alternate, c.alternateEffects);
+      let negation = AbstractValue.createFromUnaryOp(realm, "!", c.joinCondition);
+      invariant(negation instanceof AbstractValue);
+      let pathCondition = [negation];
```

Comment:
c.alternate may itself be a PossiblyNormalCompletion and, if so, you should append its pathConditions to get the path that applies to the normal branch.

Reply:
How so? JoinedAbruptCompletions has two AbruptCompletions as children, they cannot be PossibleNormal can they?

Reply:
Ah yes, you are quite right. I take that back.